### PR TITLE
#2767: Created data_attributes.rst and linked it in the README.md

### LINKED ISSUE #2767: Create 'data_attributes.rst'
We need to remove the following snippet from the main `README.md`, and place it in the yet to be created `documentation/programmatic/data_attributes.rst`:

>Some additional sample files have been provided, which outline how the data attribute implement should be implemented, with respect to the above post implementation:

>
-    SVM datasets
  -      dataset url
  -      file upload
-    SVR datasets
  -      dataset url
  -      file upload

>**Note:** the content of each of the above files, can substituted for the above data attribute.
Data Attributes

>The following (non-exhaustive) properties define the above implemented data attribute:

>
  -  `model_id`: the numeric id value, of the generated model in the nosql datastore
  -  `model_type`: corresponds to the desired model type, which can be one of the following:
    - `classification`
    - `regression`
  -  session_id: the numeric id value, that represents the dataset stored in the sql database.
  -  dataset_type: corresponds to one of the following types:
      -  dataset_url: indication that the supplied dataset will be url references
      -  file_upload: indication that the supplied dataset(s) will be defined as a json string within the dataset attribute
 -   session_type: corresponds to one of the following session types:
      -  data_new
      -  data_append
      -  model_generate
      -  model_predict
 -   svm_dataset_type: corresponds to one of the following dataset types:
      -  json_string: indicate that the dataset is being sent via a post request
 -   sv_kernel_type: the type of kernel to apply to the support vector model_type:
      -  linear
      -  polynomial
      -  rbf
      -  sigmoid
 -   prediction_input[]: an array of prediction input, supplied to the generated model to compute a prediction

Then, in place of the above removed snippet, we can link to the rst file (to be created), in the `README.md`.

### PR DESCRIPTION
**Pull request by rahul-kumi**
PR submitted to resolve #2767.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -309,52 +309,9 @@ requests.post(endpoint_url, headers=headers, data=json_string_here)
...
-  - [dataset url](https://github.com/jeff1evesque/machine-learning/tree/master/interface/static/data/json/programmatic_interface/svm/dataset_url)
-  - [file upload](https://github.com/jeff1evesque/machine-learning/tree/master/interface/static/data/json/programmatic_interface/svm/file_upload)
-- [SVR datasets](https://github.com/jeff1evesque/machine-learning/blob/master/interface/static/data/json/programmatic_interface/svr)
-  - [dataset url](https://github.com/jeff1evesque/machine-learning/tree/master/interface/static/data/json/programmatic_interface/svr/dataset_url)
-  - [file upload](https://github.com/jeff1evesque/machine-learning/tree/master/interface/static/data/json/programmatic_interface/svr/file_upload)
-
-**Note:** the content of each of the above files, can substituted for the above
- `data` attribute.
-
 #### Data Attributes
```
Comment (jeff1evesque) [inline README.md]: Remove the `#### Data Attributes` header.

**Comment by jeff1evesque** [comment]
In addition to the [above request](https://github.com/jeff1evesque/machine-learning/pull/2769#pullrequestreview-3660539), could you move `documentation/programmatic/data_attributes.rst` to `documentation/programmatic_interface/data_attributes.rst`:

```bash
$ cd /path/to/machine-learning
$ git checkout feature-2767
## make the above change
$ git add README.md
$ git commit -m "#2767: README.md, remove unnecessary header"
## move file into adjusted directory
$ git mv documentation/programmatic/data_attributes.rst documentation/programmatic_interface/data_attributes.rst
$ git commit -m "#2767: data_attributes.rst, move into proper directory"
$ git push origin feature-2767
```

Other than that, the pull request looks good!  My apologies for not giving the proper URL for `data_attributes.rst`, during the initial requests.

**Comment by jeff1evesque** [comment]
I changed your above pull request body from:

>PR submitted to resolve issue: #2767 and #2768 .

to the following:

>PR submitted to resolve #2767.

This way, when the [pull request](https://github.com/jeff1evesque/machine-learning/blob/master/contributing.md#pull-request-body) is merged, #2767 will automatically close.

**Comment by jeff1evesque** [comment]
Still need to remove the older `data_attributes.rst`:

```bash
$ git rm /path/to/machine-learning/documentation/programmatic/data_attributes.rst
$ git commit -m "#2767: data_attributes.rst, remove duplicate file"
$ git push origin feature-2767
```

**Comment by rahul-kumi** [comment]
sorry about that @jeff1evesque , I missed a git add statement after the removal.

**Comment by rahul-kumi** [comment]
Any other issue that I could work on @jeff1evesque ?

**Comment by jeff1evesque** [comment]
@rahul-kumi, seems we messed up a little bit with the syntax for our commited [`data_attributes.rst`](https://github.com/jeff1evesque/machine-learning/blob/master/documentation/programmatic_interface/data_attributes.rst).  Could you open up a new issue, and adjust the syntax.  The following resources will guide you on how to adjust for the rst syntax highlighting:

- https://gist.github.com/dupuy/1855764
- https://raw.githubusercontent.com/scikit-learn/scikit-learn/master/doc/install.rst
  - corresponds to https://github.com/scikit-learn/scikit-learn/blob/master/doc/install.rst

In addition to the new issue, I'd move `#### Data Attributes` to the top of the rst file.

**Comment by jeff1evesque** [comment]
Remember when you create a new issue, to commit to the right branch:

```bash
$ git checkout master
$ git pull upstream master
$ git pull origin master
$ git checkout -b bug-[REPLACE-WITH-ISSUE-#]
## make your changes
$ git add /path/to/machine-learning/documentation/programmatic_interface/data_attributes.rst
$ git commit -m "#[REPLACE-WITH-ISSUE-#]: data_attributes.rst, [your own comment]"
$ git push origin bug-[REPLACE-WITH-ISSUE-#]
```

**Comment by jeff1evesque** [comment]
@rahul-kumi, if you have time, you can create another issue.  We can move the following from the main `README.md` into `documentation/configuration/setup_clone.rst`:

>
### Simple clone
>
```bash
cd /[destination-directory]
sudo git clone https://[account]@github.com/[account]/machine-learning.git
cd machine-learning
git remote add upstream https://github.com/[account]/machine-learning.git
```
>
**Note:** `[destination-directory]` corresponds to the desired directory path,
 where the project repository resides.  `[account]` corresponds to the git
 username, where the repository is being cloned from.  If the original
 repository was forked, then use your git username, otherwise, use
 `jeff1evesque`.
>
### Commit hash
>
```bash
cd /[destination-directory]
sudo git clone https://[account]@github.com/[account]/machine-learning.git
cd machine-learning
git remote add upstream https://github.com/[account]/machine-learning.git
# stop vagrant
vagrant halt
# ensure diffs don't prevent checkout, then checkout hash
git checkout -- .
git checkout [hash]
```
>
**Note:** the hashes associated with a release, can be found under the
 corresponding tag value, on the [release](https://github.com/jeff1evesque/machine-learning/releases)
 page.
>
**Note:** `[destination-directory]` corresponds to the desired directory path,
 where the project repository resides.  `[account]` corresponds to the git
 username, where the repository is being cloned from.  If the original
 repository was forked, then use your git username, otherwise, use
 `jeff1evesque`.
>
### Release tag
>
```bash
cd /[destination-directory]
# clone release tag: master branch does not exist
sudo git clone -b [release-tag] --single-branch --depth 1 https://github.com/[account]/machine-learning.git [destination-directory]
git remote add upstream https://github.com/[account]/machine-learning.git
# create master branch from remote master
cd machine-learning
git checkout -b master
git pull upstream master
# return to release tag branch
git checkout [release-tag]
```
>
**Note:** `[release-tag]` corresponds to the [release tag](https://github.com/jeff1evesque/machine-learning/tags)
 value, used to distinguish between releases.
>
**Note:** `[destination-directory]` corresponds to the desired directory path,
 where the project repository resides.  `[account]` corresponds to the git
 username, where the repository is being cloned from.  If the original
 repository was forked, then use your git username, otherwise, use
 `jeff1evesque`.

Then, like #2767, you can add a `Note:` reference, in the main `README.md`, to the above rst file.  Also, you'll need to adjust the links directly under the `Configuration` header in the main `README.md`, to the headers of the `setup_clone.rst`.

**Comment by jeff1evesque** [comment]
@rahul-kumi, I will add an [rst linter](https://pypi.python.org/pypi/restructuredtext_lint) sometime later tonight.  This will help us catch errors, during pull request, if any of our rst markdown files, does not conform to a standardized syntax guide.

**Comment by rahul-kumi** [comment]
@jeff1evesque, I will raise two issues, but if you create them, if you don't mind could you add the label 'hacktoberfest' to them....

**Comment by jeff1evesque** [comment]
@rahul-kumi, I'm not sure if it's worth adding the tag just for 1-5 issues.  Beyond your contribution(s), I'm not sure how I'd feel with others, given the current issues, as well as the same reasoning as the [scikit-learn](https://www.mail-archive.com/scikit-learn@python.org/msg00563.html) project.  If at any time someone contributes, that's no problem, and I am willing to review the PR as they come.

**Pull request merged**